Dockerfile.template

```diff
@@ -19,9 +19,12 @@ ENV _BASH_LATEST_PATCH {{ .patch.version }}
 {{ ) end -}}
 # prefixed with "_" since "$BASH..." have meaning in Bash parlance
 
+# SOURCE_DATE_EPOCH is consumed by build scripts
+ARG SOURCE_DATE_EPOCH
+
```

Comment:
The rest of this is unobjectionable, but I'm not in favor of `ARG`.

```suggestion
```

Reply:
Sorry, could you explain the problem?
Isn't it quite common to specify the epoch for repro builds?

Reply:
The `SOURCE_DATE_EPOCH` has been already used in the meta scripts, no?

Reply:
@tianon Please let me know how I can make progress on this PR (and other PRs tracked in https://github.com/docker-library/official-images/issues/16044) 🙏 

Do you prefer the `SOURCE_DATE_EPOCH` to be set to the timestamp of `bash.tar.gz`? ( As in [golang](https://github.com/docker-library/golang/blob/ea6bbce8c9b13acefed0f5507336be01f0918f97/Dockerfile-linux.template#L107 ))

I can update this PR like that if you like, but I think having the "global" `ARG SOURCE_DATE_EPOCH` here still makes sense, as `SOURCE_DATE_EPOCH` also applies to [`apk`](https://gitlab.alpinelinux.org/alpine/apk-tools/-/commit/22be715f530aa81cc522899d95ef4084f5ef8a56) commands, etc.

Reply:
@tianon @tonistiigi

Updated the PR to take `SOURCE_DATE_EPOCH` from the source material (as in [`golang`](https://github.com/docker-library/golang/blob/ea6bbce8c9b13acefed0f5507336be01f0918f97/Dockerfile-linux.template#L107)):
```bash
SOURCE_DATE_EPOCH="$(find /usr/src/bash -type f -exec stat -c '%Y' {} + | sort -nr | head -n1)"
```

Let me know if this solves your concern.

Reply:
Completely removed `ARG SOURCE_DATE_EPOCH`

Reply:
Is this (and other PRs in https://github.com/docker-library/official-images/issues/16044) mergeable? 🙏 @tianon @tonistiigi